### DISCUSSION TITLE
How to level out the camera after breaking a tether

### DISCUSSION THREAD

**Discussion by elbarto1980** [pelicanmapping/osgearth#2380] [Support Q&A] [answered]
Hi,
I have a question about how to restore the correct behavior of the EarthManipulator after having tethered it to a node, using the TETHER_CENTER_AND_ROTATION mode.
When I break the tether using the clearViewpoint() method, the camera is left in a weird state and it is not controllable. To restore its correct behavior, I have to change the tether mode to TETHER_CENTER, wait for the camera to be "realigned" and only then I can break the tether.

I saw in the the EarthManipulator.cpp file that it does that [here](https://github.com/gwaldron/osgearth/blob/2b61f0e530e0bab78ec4d631c08b0cbd5bc1ec46/src/osgEarth/EarthManipulator.cpp#L2004C20-L2004C20), but I can't figure out how to do that in my code, outside the EarthManipulator class.

You can reproduce the issue using the osgeart_manip demo application if you break the tether when the TETHER_CENTER_AND_ROTATION mode is active.

What is the best way to restore the manipulator when this happens?

Thank you for your help!

P.S.: These are my caps...
```
[osgEarth]  [Registry] Note: GDAL_DATA environment variable is not set
[osgEarth]  [Capabilities] Capabilities: 
[osgEarth]  [Capabilities]   osgEarth Version:  3.2.0 build 113
[osgEarth]  [Capabilities]   OSG Version:       3.6.5
[osgEarth]  [Capabilities]   GDAL Version:      3.0.4
[osgEarth]  [Capabilities]   GEOS Version:      3.8.0
[osgEarth]  [Capabilities]   GPU Vendor:        NVIDIA Corporation
[osgEarth]  [Capabilities]   GPU Renderer:      NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2
[osgEarth]  [Capabilities]   GL/Driver Version: 4.6.0 NVIDIA 535.129.03
[osgEarth]  [Capabilities]   GL Core Profile:   no
```

**Comment by gwaldron** [comment, marked as answer]
We probably need to add that same re-alignment code whenever a tether breaks. I will convert this to an Issue. Thanks.

**Reply by elbarto1980** [reply]
Thank you!
Actually, there is a `resetLookAt()` in the [clearViewpoint()](https://github.com/gwaldron/osgearth/blob/2b61f0e530e0bab78ec4d631c08b0cbd5bc1ec46/src/osgEarth/EarthManipulator.cpp#L1313) method, but it is commented out.
In the meantime, I have been able to mitigate the issue by setting a viewpoint that has a focalpoint in `earthManipulator->centerMap()` and a range equal to `viewer->getCamera()->getViewport()->height()`, after the `clearViewpoint()`.
